sw/host/opentitanlib/src/app/config/hyperdebug_teacup.json

```diff
@@ -0,0 +1,417 @@
+      "alias_of": "CN9_19"
+    },
+    {
+      "name": "IOB12",
+      "alias_of": "CN9_21"
+    },
+    {
+      "name": "IOC0",
+      "alias_of": "CN9_13"
+    },
+    {
+      "name": "IOC1",
+      "alias_of": "CN9_15"
+    },
+    {
+      "name": "IOC2",
+      "alias_of": "CN9_17"
+    },
+    {
+      "name": "IOC3",
+      "alias_of": "CN9_6"
+    },
+    {
+      "name": "IOC4",
+      "alias_of": "CN9_4"
+    },
+    {
+      "name": "IOC5",
+      "alias_of": "CN8_14"
+    },
+    {
+      "name": "IOC6",
+      "alias_of": "CN10_7"
+    },
+    {
+      "name": "IOC7",
+      "alias_of": "CN10_18"
+    },
+    {
+      "name": "IOC8",
+      "alias_of": "CN8_16"
+    },
+    {
+      "name": "IOC9",
+      "alias_of": "CN10_31"
+    },
+    {
+      "name": "IOC10",
+      "alias_of": "CN10_33"
+    },
+    {
+      "name": "IOC11",
+      "alias_of": "CN10_2"
+    },
+    {
+      "name": "IOC12",
+      "alias_of": "CN10_4"
+    },
+    {
+      "name": "IOR0",
+      "alias_of": "CN7_7"
+    },
+    {
+      "name": "IOR1",
+      "alias_of": "CN7_5"
+    },
+    {
+      "name": "IOR2",
+      "alias_of": "CN7_3"
+    },
+    {
+      "name": "IOR3",
+      "alias_of": "CN7_1"
+    },
+    {
+      "name": "IOR4",
+      "alias_of": "CN7_16"
+    },
+    {
+      "name": "IOR5",
+      "alias_of": "CN9_14"
+    },
+    {
+      "name": "IOR6",
+      "alias_of": "CN9_16"
+    },
+    {
+      "name": "IOR7",
+      "alias_of": "CN9_18"
+    },
+    {
+      "name": "IOR8",
+      "alias_of": "CN9_20"
+    },
+    {
+      "name": "IOR9",
+      "alias_of": "CN9_22"
+    },
+    {
+      "name": "IOR10",
+      "alias_of": "CN9_24"
+    },
+    {
+      "name": "IOR11",
+      "alias_of": "CN9_26"
+    },
+    {
+      "name": "IOR12",
+      "alias_of": "CN9_28"
+    },
+    {
+      "name": "IOR13",
+      "alias_of": "CN9_30"
+    },
+    {
+      "name": "CC1",
+      "alias_of": "CN7_9"
+    },
+    {
+      "name": "CC2",
+      "alias_of": "CN7_10"
+    },
+    {
+      "name": "SPI_DEV_SCK",
+      "mode": "Alternate",
+      "alias_of": "CN10_24"
+    },
+    {
+      "name": "SPI_DEV_D0",
+      "mode": "Alternate",
+      "alias_of": "CN10_23"
+    },
+    {
+      "name": "SPI_DEV_D1",
+      "mode": "Alternate",
+      "alias_of": "CN10_10"
+    },
+    {
+      "name": "SPI1_SCK",
+      "mode": "Input",
+      "alias_of": "CN7_15"
+    },
+    {
+      "name": "SPI1_MOSI",
+      "mode": "Input",
+      "alias_of": "CN7_14"
+    },
+    {
+      "name": "SPI1_MISO",
+      "mode": "Input",
+      "alias_of": "CN7_12"
+    },
+    {
+      "name": "SPI1_CSB",
+      "mode": "Input",
+      "pull_mode": "PullUp",
+      "alias_of": "IOA7"
+    },
+    {
+      "name": "CC1",
+      "mode": "AnalogOutput",
+      "volts": 0.0
+      "alias_of": "CN7_9"
+    },
+    {
+      "name": "CC2",
+      "mode": "AnalogOutput",
+      "volts": 0.0
+      "alias_of": "CN7_10"
+    },
+    {
+      "name": "SW_STRAP0_WEAK",
```

Comment:
Why is there a `_WEAK` suffix for the SW straps?

Reply:
The SW straps are designed to have four different values per pin.  Either strongly driven high or low, or weakly (~1MOhm) pulled up or down.  On production hardware this will be achieve through physical resistors to a power rail.  For simulation when connected to HyperDebug, the each SW strap pin of OpenTitan is simultaneously connected to two HyperDebug pins, one of them through a 1M resistor.  That is, SW_STRAP0 and SW_STRAP0_WEAK are two separate pins on HyperDebug, and the former will be configured as either `PushPull/true`, `PushPull/false` or `Input`, with the last option allowing the pulling the SW strap weakly, by setting `SW_STRAP0_WEAK' either high or low.